### PR TITLE
implement probabilistic forecast using GEFS

### LINKED ISSUE #54: probabilistic reference forecasts from NWP
Some options include:

* statistics using historical data
* Global Ensemble Forecast System cloud cover. 0.5 deg, 21 members, 7+ days. [GEFS](https://nomads.ncep.noaa.gov/txt_descriptions/GFS_Ensemble_high_resolution_doc.shtml)
* HRRR Ensemble irradiance or cloud cover. Not operational. Perhaps available from ESRL ftp.
* High Resolution Ensemble Forecast System. 3 km, 8 members, 36 hrs. [HREF](https://nomads.ncep.noaa.gov/txt_descriptions/HREF_doc.shtml). [AMS 2018](https://ams.confex.com/ams/98Annual/webprogram/Paper334406.html)
* Short Range Ensemble Forecast System. 16 km, 21 members, 3 days. [SREF](https://nomads.ncep.noaa.gov/txt_descriptions/SREF_doc.shtml)?

I won't start implementing this until #10 and #53 are complete.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #54 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

To do:

- [x] decide format of returned `weather_fx_ens`
- [x] implement `main.run_probabilistic_nwp`, including functionality to convert deterministic ensemble into probabilistic forecast
- [x] figure out copy pasta in `run_probabilistic_nwp`

### CODE REVIEW

**Comment by wholmgren** [comment]
@alorenzo175 ready for initial review. I’ve plotted the outputs and they seem reasonable. 

I’m not sure that the new run_probabilistic_nwp function is needed. The only difference now is how it handles ac power. That could be fixed by modifying our pvpower functions to create dni_extra with same shape as input irradiance. are there other reasons to want a separate run function?

**Comment by wholmgren** [comment]
Obviously the copy pasta would need to be reduced if we keep the second function.

File: solarforecastarbiter/io/nwp.py [hunk 1]
```diff
@@ -71,6 +71,7 @@ def load_forecast(
             * 'gfs_0p25'
             * 'rap'
             * 'nam_12km'
+            * 'gefs_p[01-20].nc', 'gefs_c00', 'gefs_avg', or 'gefs_spr'
```
Comment (alorenzo175) [inline solarforecastarbiter/io/nwp.py]: .nc?

File: solarforecastarbiter/reference_forecasts/models.py [hunk 2]
```diff
@@ -315,6 +309,132 @@ def _adjust_gfs_start_end(start, end):
...
+    ProbabilisticForecast with ``axis='x'`` and
+    ``constant_values=[0, 5, ...95, 100].
+    """
+    start_floored, end_ceil = _adjust_gfs_start_end(start, end)
+
+    def _load_gefs_member(member, solar_position):
+        cloud_cover_mixed, air_temperature, wind_speed = load_forecast(
+            latitude, longitude, init_time, start_floored, end_ceil, member,
+            variables=('cloud_cover', 'air_temperature', 'wind_speed'))
+        cloud_cover = _unmix_various_gfs_intervals(
```
Comment (alorenzo175) [inline solarforecastarbiter/reference_forecasts/models.py]: gfs or gefs?

**Comment by wholmgren** [comment]
There's an issue with start times between the init time and the first forecast time (3h later) that I'd like to address as part of #186 once this is merged. Otherwise I think it's good.

File: solarforecastarbiter/reference_forecasts/tests/test_models.py [hunk 3]
```diff
@@ -41,21 +41,21 @@ def test_default_load_forecast_failures(model):
...
+        assert isinstance(o, (pd.Series, pd.DataFrame))
         assert o.index[0] == start
         if end_strict:
             assert o.index[-1] == end
         else:
             assert o.index[-1] <= end
     # check irradiance limits
     for o in out[0:3]:
-        assert (o >= 0).all() and (o < 1300).all()
+        assert (o >= 0).all().all() and (o < 1300).all().all()
```
Comment (alorenzo175) [inline solarforecastarbiter/reference_forecasts/tests/test_models.py]: cool that this works now on Series.

**Pull request merged**